101-rds-deployment-full-ha/azuredeploy.json

```diff
@@ -0,0 +1,1180 @@
+                            },
+                            "name": "[concat(parameters('vmName'),copyIndex(1),'pip')]",
+                            "copy": {
+                                "name": "[concat(parameters('vmName'),'pipcopy')]",
+                                "count": "[parameters('loopCount')]"
+                            }
+                        },
+                        {
+                            "type": "Microsoft.Network/networkInterfaces",
+                            "apiVersion": "2019-12-01",
+                            "location": "[parameters('location')]",
+                            "dependsOn": [
+                                "[concat(parameters('vmName'),copyIndex(1),'pip')]"
+                            ],
+                            "properties": {
+                                "ipConfigurations": [
+                                    {
+                                        "name": "ipconfig",
+                                        "properties": {
+                                            "subnet": {
+                                                "id": "[parameters('subNetRef')]"
+                                            },
+                                            "privateIPAllocationMethod": "[if(contains(parameters('vmName'),'dc'),if(equals(copyIndex(1),1),'static','dynamic'),'dynamic')]",
+                                            "privateIPAddress": "[if(contains(parameters('vmName'),'dc'),if(equals(copyIndex(1),1),parameters('firstDcIP'),json('null')),json('null'))]",
+                                            "publicIPAddress": {
+                                                "id": "[resourceId('Microsoft.Network/publicIPAddresses',concat(parameters('vmName'),copyIndex(1),'pip'))]"
+                                            },
+                                            "loadBalancerBackendAddressPools": "[if(not(empty(parameters('intLbBackEndPool'))),if(not(empty(parameters('pubLbBackEndPool'))),variables('pubIntlbPool'),variables('intlbPool')),json('null'))]"
+                                        }
+                                    }
+                                ]
+                            },
+                            "name": "[concat(parameters('vmName'),copyIndex(1),'nic1')]",
+                            "copy": {
+                                "name": "[concat(parameters('vmName'),'niccopy')]",
+                                "count": "[parameters('loopCount')]"
+                            }                                                        
+                        },
+                        {
+                            "type": "Microsoft.Compute/virtualMachines",
+                            "apiVersion": "2019-07-01",
+                            "location": "[parameters('location')]",
+                            "dependsOn": [
+                                "[concat(parameters('vmName'),copyIndex(1),'nic1')]"

```

Comment:
as above, check all of your dependsOn properties and use resourceId()  

won't comment on the rest